packages/react-native-editor/__device-tests__/README.md

```diff
@@ -22,9 +22,23 @@ TBA
 
 TLDR; to run the tests locally ensure metro isn't running and then run `npm run native test:e2e:ios:local` and `npm run native test:e2e:android:local` for the desired platform.
 
-Those commands include the process to build a testable version of the app, if it's the case you don't want to run the 
-full suite and want to run a specific file or files you can use the e2e build commands that can be found in the package.json for the respective platform and then 
-run `TEST_RN_PLATFORM=android npm run native device-tests <pattern>` where the pattern can just be the file path.
+Those commands include the process to build a testable version of the app with these steps:
+1. Create the JS bundle via `test:e2e:bundle:(ios|android)` 
+1. Compile the native app code via `test:e2e:build-app:(ios|android)`
+1. Call the test runner via `device-tests:local`
+
+Once the JS bundle and native app code are created they can be re-used by the test runner `device-tests:local` in subsequent runs. While writing tests: 
+- If you only changed the native app code you can run only `test:e2e:build-app:(ios|android)` followed by `device-tests:local`
+- If you only changed JS app code you can run only `test:e2e:bundle:(ios|android)` followed by `device-tests:local`
+- If you didn't change native or JS app code but only have modified e2e tests under `__device-tests__` you only need to re-rerun `device-tests:local` which uses the pre-built native and JS app code
+- If it's the case you don't want to run the 
+full suite and want to run a specific file or files you can run `TEST_RN_PLATFORM=android npm run native device-tests <pattern>` where the pattern can just be the file name.
```

Comment:
Thanks for adding these tips! I'd found them very useful.

Reply:
I've found that local tests are not working on Xcode 12, but work on Xcode 11.6. I've come across some issues here:
- [Cannot launch WebDriverAgent : error - 'carthage bootstrap --platform iOS\,tvOS'](https://github.com/appium/appium/issues/13532)
- [appium 1.18.1 and Xcode 12 beta 6 - Building scheme "iOS Framework" in CocoaAsyncSocket.xcodeproj failure](https://github.com/appium/appium/issues/14703)
- The fix looks like it might need a beta version of Appium: https://github.com/appium/appium/issues/14703#issuecomment-690322194

I wonder if you've come across this, but since we're still using Xcode 11 ([here](https://github.com/WordPress/gutenberg/blob/a9267fe16cc4138b945ef9b88a7b8ca7ae832af6/.github/workflows/rnmobile-ios-runner.yml#L55)), it might be worth noting something like this in the "Running the tests locally" section:

> On iOS, execution of E2e tests is officially supported on Xcode 11, support for Xcode 12 is in the works. If you have Xcode 12 and Xcode 11 installed, switch to Xcode 11 before running tests, e.g. `sudo xcode-select -switch /Applications/Xcode11.app`.

It's up to you though, definitely not a blocker.

Reply:
Thanks for testing locally @guarani! Will switch to Xcode 12 for CI in GitHub Actions and there it's very similar to how it's ran locally so I'll try to fix these issue in this PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2744